docs/gc.md

```diff
@@ -141,72 +141,90 @@ Object ages are tracked for each region with a maximum of 24 possible generation
+The *eden* space is a set of regions of age 0, which contain the newest objects allocated. When the *eden* space is full, a partial GC cycle runs to reclaim space, typically by using a *copy forward* operation. In specific cases, a *mark-compact* operation might be used, for example, when there are not enough free survivor regions available. The partial GC cycle is a *stop-the-world* (STW) operation, after which new regions are allocated to create a new *eden* space. A partial cycle always includes the *eden* space, but might include older regions. Objects from collectible regions of age N are moved into another region of the same age N or to an empty region that is assigned an age of N. Then, the ages of all regions across the heap are incremented by 1, except for the maximum age 24 regions. Regions of age 24 are included in partial GC collection sets in order to defragment them.
 
-Over time, transient objects drop out of the live set and viable survivor regions become scarce. This problem is solved by introducing a global mark phase (GMP) to mark and sweep the entire heap. GMP is a hybrid *incremental-concurrent* GC cycle. Each GMP increment is interleaved with a PGC cycle so that regions that are left sparse after sweeping the heap can be collected, cleared, and recycled during and after the GMP.
+Partial GC cycles work to reclaim free regions in the heap for allocating new objects. Because some objects from *eden* regions always survive, a partial GC cycle can reclaim only about  90% of this memory. To keep up with object allocation, partial GC cycles also reclaim free regions by defragmenting older regions. For example, a partial GC cycle that moves objects from 5 fragmented older regions into 2 empty regions, reclaims 3 regions for new object allocation. However, over time the overall amount of fragmented memory decreases and records about object liveness in older regions becomes less accurate. Eventually, the work done by partial GC cycles to reclaim memory cannot keep pace with memory allocation. Free regions become so scarce that a global *mark* operation (GMP) is required to build a new record of object liveness across the heap. A *sweep* operation uses this record to measure the amount of free memory in fragmented older regions, which later partial GC cycles can act upon to move objects and reclaim free regions.
```

Comment:
grammar: records become (not becomes)

Perhpas it's more generic to say: .... cannot keep pace with memory consumption (rather than allocation). We are not talking just about mutator allocation, but also empty survivor region consumption by PGC or in other words difference between mutator allocation minus the recovered memory. In the example above, we did not really allocate 5 regions, but we also reclaimed 3, so the net is 2 regions consumed.